texar/data/data/multi_aligned_data.py

```diff
@@ -0,0 +1,586 @@
+                        },
+                        'image_options': {
+                            'image_feature_name': 'image',
+                            'resize_height': 512,
+                            'resize_width': 512,
+                        },
+                        'data_name': 't',
+                    }
+                ]
+                'batch_size': 1
+            }
+            data = MultiAlignedData(hparams)
+            iterator = DataIterator(data)
+            for batch in iterator:
+                # batch contains the following
+                # batch_ == {
+                #    'x_text': [['<BOS>', 'NewYork', 'City', 'Map', '<EOS>']],
+                #    'x_text_ids': [['1', '100', '80', '65', '2']],
+                #    'x_length': [5],
+                #
+                #    # "t_image" is a list of a "numpy.ndarray" image
+                #    # in this example. Its width equals to 512 and
+                #    # its height equals to 512.
+                #    't_image': [...]
+                # }
+
+    """
+
+    def __init__(self, hparams, device: Optional[torch.device] = None):
+        self._hparams = HParams(hparams, self.default_hparams())
+        self._other_transforms: List[
+            List[Callable[[Union[List[str], int, float, Dict[str, Any]]],
+                          Union[List[str], int, float, Dict[str, Any]]]]] = []
+        # Defaultizes hyperparameters of each dataset
+        datasets_hparams = self._hparams.datasets
+        defaultized_datasets_hparams = []
+        for hparams_i in datasets_hparams:
+            data_type = hparams_i.get("data_type", None)
+            defaultized_ds_hpms = HParams(hparams_i,
+                                          _default_dataset_hparams(data_type))
+            defaultized_datasets_hparams.append(defaultized_ds_hpms)
+            self._other_transforms.append(
+                defaultized_ds_hpms.other_transformations)
+        self._hparams.datasets = defaultized_datasets_hparams
+
+        self._vocab = self.make_vocab(self._hparams.datasets)
+        self._embedding = self.make_embedding(self._hparams.datasets,
+                                              self._vocab)
+
+        self._name_prefix: List[str] = []
+        self._names: List[Dict[str, Any]] = []
+        datasources: List[DataSource] = []
+        filters: List[Optional[Callable[[str], bool]]] = []
+        self._databases: List[DataBase] = []
+        for idx, hparams_i in enumerate(self._hparams.datasets):
+            dtype = hparams_i.data_type
+            datasource_i: DataSource
+
+            if _is_text_data(dtype) or _is_scalar_data(dtype):
+                datasource_i = TextLineDataSource(
+                    hparams_i.files,
+                    compression_type=hparams_i.compression_type)
+                datasources.append(datasource_i)
+                if _is_text_data(dtype) and (hparams_i["length_filter_mode"] is
```

Comment:
The `is` is not yet changed. It should be `==`.

Reply:
Done! [462312cab452a7f1c3eacce634d65f1a50c2fbbc]